Sources/Imaging/Core/ImageReslice/index.js

```diff
@@ -30,6 +31,95 @@ function vtkImageReslice(publicAPI, model) {
   let indexMatrix = null;
   let optimizedTransform = null;
 
+  function getImageResliceSlabTrap(tmpPtr, inComponents, sampleCount, f) {
+    let subTmpPtr = tmpPtr;
+    let m = inComponents;
+    const n = sampleCount - 1;
+    do {
+      let result = subTmpPtr[0] * 0.5;
+      for (let k = n - 1; k !== 0; --k) {
+        subTmpPtr = tmpPtr.subarray(inComponents);
+        result += subTmpPtr[0];
+      }
+      subTmpPtr = tmpPtr.subarray(inComponents);
+      result += subTmpPtr[0] * 0.5;
+      tmpPtr.fill(result * f, 0, 1);
+      tmpPtr = tmpPtr.subarray(1);
```

Comment:
I think it should be simplified into 
```
result += tmpPtr[inComponents] * 0.5;
tmpPtr[0] = result * f;
tmpPtr = tmpPtr.subarray(1);
```

I wonder if there is an advantage to create subarrays when iterating one by one.
Instead you could do something like that (I have not double checked if it is the same as the c++ version):
```
for (let j = 0; j < inComponents; j += 1) {
  let result = tmpPtr[j] * 0.5;
  for (let k = 1; k < n; k += 1) {
     result += tmpPtr[j+k];
  }
  result += tmpPtr[j+n] * 0.5;  
  tmpPtr[j] = result;
}
```